packages/headless/src/test/mock-engine.ts

```diff
@@ -78,7 +80,11 @@ function buildMockEngine<T extends AppState>(
   return {
     mockStore: store,
     store,
-    state: mockState(),
+    state: {
+      ...mockState(),
+      searchHub: getSearchHubInitialState(),
+      pipeline: getPipelineInitialState(),
+    },
```

Comment:
This is not the cleanest solution, but I did it to fix the build and keep the PR scoped. I will work on streamlining the mock-engine file in the next PR.

One thing it reveals though is the `ProductRecommendations` request does not leverage the `pipeline` option. @francoislg, I'm wondering if this is by design, or just overlooked?

Reply:
By design, recommendations really don't need pipelines :P 

I prefer not putting them and adding it back if needed than removing it when we create the API.

Reply:
okay, I see. Atm, when creating an engine, it's possible to specify a `pipeline`. This option is used by the search and regular recommendation requests, but not by product recommendations.

It's not a big issue, but I could see someone in the future opening a maintenance case reporting that their configured pipeline is not being sent.

Instead, it might be better to create a dedicated engine for product recommendations, not exposing the `pipeline` option. I want to think more about `Recommendations` and `Product Recommendations`. For now, I will leave things as they are.